toolchain/check/testdata/facet/access.carbon

```diff
@@ -80,6 +80,55 @@ fn UseIndirect[T:! I](x: T) -> T {
 //@dump-sem-ir-end
 }
 
+// --- fail_todo_access_through_call.carbon
+library "[[@TEST_NAME]]";
+
+interface I {
+  let X:! type;
+  fn G() -> X;
+}
+
+fn F(U:! I where .X = .Self) {
+  U.G().G();
+
+  // TODO: This should typecheck.
+  // - The first `.` is on a NameRef of type FacetType for `I where .X = .Self`.
+  //   - This finds `G` through the FacetType.
+  // - The second `.` is on a Call of type FacetAccessType into `BindSymbolicName` with type FacetType for `I`.
+  //   - This finds `G` through the FacetType (impl lookup strips off FacetAccessType).
+  // - The second `.` is on a Call of type FacetAccessType into `ImplWitnessAccess` of `I.X` into `LookupImplWitness`, which has type `type`
+  //   - Can't make calls on an `ImplWitnessAccess`.
+  //
+  // CHECK:STDERR: fail_todo_access_through_call.carbon:[[@LINE+4]]:3: error: type `.(I.X)` does not support qualified expressions [QualifiedExprUnsupported]
+  // CHECK:STDERR:   U.G().G().G();
+  // CHECK:STDERR:   ^~~~~~~~~~~
+  // CHECK:STDERR:
+  U.G().G().G();
+}
+
+// --- fail_todo_access_through_call_converts_is_a_type.carbon
+library "[[@TEST_NAME]]";
+
+interface I {
+  let X:! type;
+  fn G() -> X;
+}
+
+fn F(U:! I where .X = .Self) {
+  // TODO: The return type of G() is a FacetAccessType of a BindSymbolicName. We
+  // try to convert a Call of type FacetAccessType (of type `type`) to `type`
+  // which fails.
+  //
+  // CHECK:STDERR: fail_todo_access_through_call_converts_is_a_type.carbon:[[@LINE+7]]:3: error: cannot convert non-type value of type `.Self` to `type` with `as` [ConversionFailureNonTypeToFacet]
+  // CHECK:STDERR:   U.G() as type;
+  // CHECK:STDERR:   ^~~~~~~~~~~~~
+  // CHECK:STDERR: fail_todo_access_through_call_converts_is_a_type.carbon:[[@LINE+4]]:3: note: type `.Self` does not implement interface `Core.As(type)` [MissingImplInMemberAccessNote]
+  // CHECK:STDERR:   U.G() as type;
+  // CHECK:STDERR:   ^~~~~~~~~~~~~
+  // CHECK:STDERR:
+  U.G() as type;
```

Comment:
I think this one should fail: `U.G() as type` should be a value expression of type `U`, and we can't convert a value of type `U` to type `type`.

Reply:
Maybe `U.X as type` would be the thing we expect to work here. But that seems a little vacuous because it's already of type `type`.

Reply:
That sounds right, it's the FacetAccessType that is really messing with me.

Specifically, the return type here is `FacetAccessType(Special) of U` which knows U and knows the *type of U*. So then I want to treat the return as-if it's a value of type of the *type of U* which is a FacetType.

But that's wrong.. but I am having trouble finding where I fell off the path.

Reply:
Ok so I think I get it...

A facet value is a type (or a placeholder for a type) already, it just also holds a witness to say what interface(s) it's accessing.

So FacetAccessType isn't doing a type-of operation, it's just passing along the type that the facet value _is_. So the `FacetAccessType of U` is just `U` still, but asking for the final type that arrives in U.